db/migrate/20130920225539_create_users.rb

```diff
@@ -1,15 +1,43 @@
 class CreateUsers < ActiveRecord::Migration
   def change
-    enable_extension "hstore"
-
     create_table :users do |t|
       t.string :name
-      t.string :email
+      t.string :email, null: false, default: ""
       t.text :bio
-      t.hstore :demographics
       t.boolean :admin, default: false
 
       t.timestamps
+
+      ## Database authenticatable
+      t.string :encrypted_password, null: false, default: ""
+
+      ## Recoverable
+      t.string   :reset_password_token
+      t.datetime :reset_password_sent_at
+
+      ## Rememberable
+      t.datetime :remember_created_at
+
+      ## Trackable
+      t.integer  :sign_in_count, default: 0, null: false
+      t.datetime :current_sign_in_at
+      t.datetime :last_sign_in_at
+      t.inet     :current_sign_in_ip
+      t.inet     :last_sign_in_ip
+
+      #Omniauth Single Service Authentication
+      t.string :provider
+      t.string :uid
+
+      t.string :confirmation_token
+      t.datetime :confirmed_at
+      t.datetime :confirmation_sent_at
+      t.string :unconfirmed_email # Only if using reconfirmable
+
     end
+
```

Comment:
For missing indexes, email seems obvious and I'm thinking provider + uid for oauth logins.